python/lsst/dax/imgserv/api_soda.py

```diff
@@ -124,45 +143,272 @@ def img_capabilities():
+    """
+    ar = app_celery.AsyncResult(job_id)
+    phase = map_phase_from_state[ar.state]
+    e = ar.get() if phase == "ERROR" else "NONE"
+    return _uws_job_response_plain("ERROR="+e)
+
+
+@image_soda.route("/async-jobs/<job_id>/quote", methods=["GET"])
+def img_async_job_quote(job_id: str):
+    """ Get the quote for the job.
+
+    Parameters
+    ----------
+    job_id : `str`
+    Returns
+    -------
+    xml : `str`
+        the quote info for the job.
+    """
+    return _uws_job_response_plain("INFO=NOT IMPLEMENTED")
+
+
+@image_soda.route("/async-jobs/<job_id>/results", methods=["GET"])
+def img_async_job_results(job_id: str):
+    """ Get the results for the job.
+
+    Parameters
+    ----------
+    job_id : `str`
+
+    Returns
+    -------
+    xml : `str`
+        the job results.
+    """
+    # For now redirect to single result
+    return redirect(url_for('api_image_soda.img_async_job_results_result',
+                            job_id=job_id,
+                            _external=True))
+
+
+@image_soda.route("/async-jobs/<job_id>/results/result", methods=["GET"])
+def img_async_job_results_result(job_id: str):
+    """ Get the single result for the job.
+
+    Parameters
+    ----------
+    job_id : `str`
+
+    Returns
+    -------
+    xml : `str`
+        the job results.
+    """
+    ar = app_celery.AsyncResult(job_id)
+    if ar.state == "SUCCESS":
+        result = ar.get()  # retrieve path to the image output
+        resp = send_file(result,
+                         mimetype="image/fits",
+                         as_attachment=True,
+                         attachment_filename=os.path.basename(result))
+        return resp
+    else:
+        return redirect(url_for('api_image_soda.img_async_jobs_job',
+                                job_id=job_id,
+                                _external=True))
+
+
+@image_soda.route("/async-jobs/<job_id>/parameters", methods=["GET", "POST"])
+def img_async_job_parameters(job_id: str):
+    """Get/Set the parameters for the job.
+
+    Parameters
+    ----------
+    job_id : `str`
+    Returns
+    -------
+     xml : `str`
+        the job parameters.
+    """
+    ar = app_celery.AsyncResult(job_id)
+    if ar.state in ["PENDING", "STARTED"]:
+        params=str(ar.args)
+    else:
+        params = "NOT AVAILABLE"
+    if ar.state == "PENDING" and request.method == "POST":
+        # TODO: update the job parameters in PENDING state
+        new_params = _getparams()
+    return _uws_job_response_plain("PARAMS="+params)
+
+
+@image_soda.route("/async-jobs/<job_id>/owner", methods=["GET"])
+def img_async_job_owner(job_id: str):
+    """ Get the owner info for the job.
+
+    Parameters
+    ----------
+    job_id : `str`
+
+    Returns
+    -------
+    xml : `str`
+        the job owner.
+    """
+    ar = app_celery.AsyncResult(job_id)
+    # ideally owner should be stored somewhere along with the job_id
+    # and assert session against that.
```

Comment:
Yes, you should store the user who started it.  Please don't just say the owner is whoever is checking.  This is really bad.

Reply:
You should be able to store this in the params, I think you can get the params of a job by job id.

Reply:
I've updated the code to store user/owner id, if available, and being able to retrieve it later, as should be.